frontend-react/docs/rs-auth-element.md

```diff
@@ -0,0 +1,29 @@
+# AuthElement
+
+Use the `AuthElement` component to restrict access to a page. Because `react-router-dom`
+no longer supports custom route components, instead, we can use this middle layer component
+to define authorized user types and required user flags. The component uses the session
+context to authorize a user. It checks:
+
+- Logged in status: if not logged in, it redirects to `/login`
+- Feature flag (optional): use `requiredFeatureFlag` prop to define the flag required
+- User type (optional): use `requiredUserType` prop to define one or many user types authorized
+
+> `MemberType.PRIME_ADMIN` is always permitted
+
+Use the `element` prop to define the page you are requiring authorization for.
+
+### Example use
+
+```typescript jsx
+import {AuthElement} from "./AuthElement";
+import {MemberType} from "./UseOktaMemberships";
+
+const Page = () => <div>I'm a page</div>
+export const PageWithAuth = () => (
+    <AuthElement
+        element={Page}
```

Comment:
Updated it. 

I also looked into the pattern a bit. React router's change doc made a good point here...
> In the section about upgrading to v5.1, we promised that we'd discuss the advantages of using regular elements instead of components (or element types) for rendering. Let's take a quick break from upgrading and talk about that now.
>
>For starters, we see React itself taking the lead here with the <Suspense fallback={<Spinner />}> API. The fallback prop takes a React element, not a component. This lets you easily pass whatever props you want to your <Spinner> from the component that renders it.

https://reactrouter.com/en/v6.3.0/upgrading/v5#advantages-of-route-element

Reply:
Seems like their use of `compnoent` is what I equate to a `ComponentAsFunction` type (aka `() => JSX.Element`). Whereas a `react element` to them is just the rendered JSX.Element

Reply:
OK. This is a nice deep dive into understanding why. Thanks!